backend/telemetry_shard/src/connection.rs

```diff
@@ -0,0 +1,195 @@
+use futures::channel::mpsc;
+use futures::{Sink, SinkExt, StreamExt};
+use tokio::net::TcpStream;
+use tokio_util::compat::TokioAsyncReadCompatExt;
+
+#[derive(Clone, Debug)]
+pub enum Message<Out> {
+    Connected,
+    Disconnected,
+    Data(Out),
+}
+
+/// Connect to a websocket server, retrying the connection if we're disconnected.
```

Comment:
It's probably specific enough to be given a more specific name, so I shall do that (it serializes/deserializes to/from bincode, but is otherwise quite generic).